src/lib/Microsoft.Health.Events/EventHubProcessor/EventProcessorClientFactory.cs

```diff
@@ -0,0 +1,52 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using Azure.Identity;
+using Azure.Messaging.EventHubs;
+using Azure.Storage.Blobs;
+using EnsureThat;
+using Microsoft.Health.Common.Auth;
+
+namespace Microsoft.Health.Events.EventHubProcessor
+{
+    public class EventProcessorClientFactory
+    {
+        private BlobContainerClient _blobContainerClient;
+
+        public EventProcessorClientFactory(BlobContainerClient blobContainerClient)
+        {
+            _blobContainerClient = EnsureArg.IsNotNull(blobContainerClient, nameof(blobContainerClient));
+        }
+
+        public EventProcessorClient CreateProcessorClient(EventProcessorClientFactoryOptions options, EventProcessorClientOptions eventProcessorClientOptions)
+        {
+            EnsureArg.IsNotNull(options);
+
+            if (options.ServiceManagedIdentityAuth)
+            {
+                var tokenCredential = new DefaultAzureCredential();
+                return new EventProcessorClient(_blobContainerClient, options.EventHubConsumerGroup, options.EventHubNamespaceFQDN, options.EventHubName, tokenCredential, eventProcessorClientOptions);
+            }
+            else if (!string.IsNullOrEmpty(options.ConnectionString))
```

Comment:
Yeah we have a backlog item - right now none of this console stuff is documented because we were planning on doing a couple more iterations before putting out some documentation just we don't have to keep changing it.